ProgettoMP2018-2019/test/warehouseTest/ShelfTest.java

```diff
@@ -0,0 +1,67 @@
+package warehouseTest;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import myTools.Product;
+import warehouse.Shelf;
+
+public class ShelfTest {
+    private Shelf testShelf;
+    private Product productTest;
+
+    @Before
+    public void initFixture() {
+        testShelf = new Shelf(5);
+    }
+
+    @Test
+    public void GivenAproductIncreaseElementsInShelf() {
+        int expectedElementsInShelf = 1;
+
+        testShelf.placeProduct(productTest);
+        int actualElementsInShelf = testShelf.elementsInShelf();
+
+        assertEquals(expectedElementsInShelf, actualElementsInShelf);
+    }
+
+    @Test
+    public void GivenLastElementChangeShelfState() {
+        testShelf.placeProduct(productTest);
+        testShelf.placeProduct(productTest);
+        testShelf.placeProduct(productTest);
+        testShelf.placeProduct(productTest);
```

Comment:
Un for è dispendioso ma forse più bello alla vista

Reply:
i test devono essere barbari, l'ho scritto per esplicitare l'operazione e non per avere problemi con gli indici ed i numeri